app/models/market_data.rb

```diff
@@ -0,0 +1,103 @@
+class MarketData
+  VALID_INDICATORS = %w(total_supply circulating_supply)
+  INITIAL_SUPPLY = BigDecimal(336 * 10**16)
+  BURN_QUOTA = BigDecimal(84 * 10**16)
+  ECOSYSTEM_QUOTA = INITIAL_SUPPLY * 0.17
+  TEAM_QUOTA = INITIAL_SUPPLY * 0.15
+  PRIVATE_SALE_QUOTA = INITIAL_SUPPLY * 0.14
+  FOUNDING_PARTNER_QUOTA = INITIAL_SUPPLY * 0.05
+  FOUNDATION_RESERVE_QUOTA = INITIAL_SUPPLY * 0.02
+
+  attr_reader :indicator, :current_time
+
+  def initialize(indicator = nil)
+    @indicator = indicator
+    @current_time = Time.zone.now
+  end
+
+  def call
+    return unless indicator.in?(VALID_INDICATORS)
+
+    send(indicator)
+  end
+
+  private
+
+  def parsed_dao
+    @parsed_dao ||= begin
+      latest_dao = Block.recent.pick(:dao)
+      CkbUtils.parse_dao(latest_dao)
+    end
+  end
+
+  def total_supply
+    result = parsed_dao.c_i - BURN_QUOTA
+
+    (result / 10**8).truncate(8)
+  end
+
+  def circulating_supply
+    result = parsed_dao.c_i - BURN_QUOTA - ecosystem_locked - team_locked - private_sale_locked - founding_partners_locked - foundation_reserve_locked
+
+    (result / 10**8).truncate(8)
+  end
+
+  def ecosystem_locked
+    first_released_time = Time.zone.parse("2020-07-01")
```

Comment:
Use UTC timezone explicitly instead of `Time.zone`

Reply:
thanks, I will fix this.